File: Benchmarking/stats/euler16/euler16.jl
#=
2^15 = 32768 and the sum of its digits is 3 + 2 + 7 + 6 + 8 = 26.

What is the sum of the digits of the number 2^1000?
=#
#Answer: 1366

#Terminal: julia euler16.jl 2 1000
#args[1] = 2
#args[2] = 1000
num = parse(Int, ARGS[1])
power = parse(Int, ARGS[2])

#Guess on needed size for all inputs.
arrSize = power
arr = zeros(Int, arrSize)
#Setting number to num^1
arr[1] = num

function calc(arr)
  sum = 0
  carry = 0
  for i = 2 : power
    for j = 1 : length(arr)
      temp = arr[j] * 2
      if temp > 9
        arr[j] = temp % 10 + carry
        carry = 1
      else
        arr[j] = temp + carry
        carry = 0
      end
    end
  end

  for i = 1 : length(arr)
    sum += arr[i]
  end
  return sum
end

calc(arr)

File: Benchmarking/stats/euler16/euler16.java
//calculate 2^1000
//args[0] = 2
//args[1] = 1000
public class euler16 {
  public static void main (String[] args) {
      int num = Integer.parseInt(args[0]);
      int power = Integer.parseInt(args[1]);

      int[] arr = new int[power];
      arr[0] = 2;
      int carry = 0;
      int sum = 0;

      for (int i = 1; i < power; i++) {
        for (int j = 0; j < arr.length; j++) {
          int temp = arr[j] * 2;
          if (temp > 9) {
            arr[j] = temp % 10 + carry;
            carry = 1;
          }
          else {
            arr[j] = temp + carry;
            carry = 0;
          }
        }
      }
      for (int i = 0; i < arr.length; i++) {
        sum += arr[i];
      }
      //System.out.println(sum);
  }
}

